packages/config/src/configDefaults.ts

```diff
@@ -336,6 +336,31 @@ export const configProps = ({
           "Don't set config.timeoutBlocks directly. Instead, set config.networks and then config.networks[<network name>].timeoutBlocks"
         );
       }
+    },
+    ensRegistry: {
+      get() {
+        let networkConfig;
+        try {
+          networkConfig = configObject.network_config;
+        } catch {
+          //if this throws, then there's no network config, whatever
+        }
+        //NOTE: in what follows I'm not just using || or ?? because I want
+        //null to be respected but undefined not to be
+        const address = [
+          //note that network-specific locations go first
+          networkConfig?.registry?.address,
+          networkConfig?.registryAddress,
+          configObject.ens?.registry?.address,
+          configObject.ens?.registryAddress
```

Comment:
This precedence resolution locks the API behavior and should be guarded by unit tests to maintain behavior to prevent introducing a subtle bug in future maintenances.

Reply:
Again, as long as specific is before generic, I don't think there's a problem (see comments earlier).  In any case I *did* add a test that tests the specific-before-generic behavior; I don't see a need to add a test for the remainder of the order and lock it in.

Reply:
That makes sense, though It would be helpful to inform the user they're in undefined territory if they set both network specific and generic ens settings.

Reply:
Oh, hm, yeah, we could throw an error in that case!  I guess I'll go do that.

Reply:
OK, I reworked the logic here to throw an error in that case.  I did use copy/paste but I think it's fine, it's all right next to each other and it's just two.